mesonbuild/options.py

```diff
@@ -168,12 +168,12 @@ def __eq__(self, other: object) -> bool:
 
     def __ne__(self, other: object) -> bool:
         if isinstance(other, OptionKey):
-            return self._to_tuple() != other._to_tuple()
+            return self._hash != other._hash
         return NotImplemented
 
     def __ge__(self, other: object) -> bool:
         if isinstance(other, OptionKey):
-            return self._to_tuple() >= other._to_tuple()
+            return self._hash == other._hash
```

Comment:
This should be for `__eq__`, not for `__ge__` !

Reply:
Ooops! Fixed now.